ci-templates/github/model-complexity-badge/action.yml

```diff
@@ -0,0 +1,72 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capellambse contributors
+# SPDX-License-Identifier: Apache-2.0
+
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+name: 'Generate Model Complexity Badge'
+description: 'An action to create and push a model complexity badge'
+
+inputs:
+  commit_msg:
+    description: 'The commit message for updating the model complexity badge'
+    required: false
+    default: "docs: Update model complexity badge"
+  output_file:
+    description: 'The file path to store the generated model badge'
+    required: false
+    default: model-complexity-badge.svg
+  entry_point:
+    description: 'The entry point of the used model'
+  capellambse_revision:
+    description: 'The revision of the py-capellambse package to install'
+    required: false
+    default: master
+  push_model_badge:
+    description: 'Whether to push the model badge changes or not'
+    required: false
+    default: "true"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: 3.11
+    - name: Generate model badge
+      shell: bash
+      env:
+        ENTRYPOINT: ${{ inputs.entry_point }}
+        OUTPUT_FILE: ${{ inputs.output_file }}
+      run: |
+        git fetch
+        git reset
+        pip install "git+https://github.com/DSD-DBS/py-capellambse.git@${{ inputs.capellambse_revision }}"
+
+        python <<EOF
+        print("-----start-python-model-badge-----")
+        import os
+        import pathlib
+        import capellambse
+
+        model = capellambse.MelodyModel(os.getenv("ENTRYPOINT"))
+        pathlib.Path(os.getenv("OUTPUT_FILE")).write_text(model.description_badge)
+        print("-----generated-python-model-badge-----")
+        EOF
+    - name: Push model badge
+      shell: bash
+      run: |
+        git add "${{ inputs.OUTPUT_FILE }}"
+        if git diff --cached --exit-code &> /dev/null || [[ "${{ inputs.push_model_badge }}" == "false" ]];then
+          exit 0
+        fi
```

Comment:
``inputs.push_model_badge`` can be checked by the runner without even having to spawn a shell. This will also prominently mark the step as "skipped" instead of "passed" with (pretty much) empty logs. This is also how it's done for the "publish to PyPI" step in our Code QA workflow:

```yaml
    - name: Push model badge
      if: inputs.push_model_badge
      run: [...]
```

This does require the input to be a boolean (not a string), so the quotes there need to be removed as well:

```diff
   push_model_badge:
-    default: "true"
+    default: true
```